src/test/cloudWatchLogs/registry/logDataRegistry.test.ts

```diff
@@ -50,6 +86,121 @@ describe('LogDataRegistry', async function () {
+            await testUpdateLog('tail')
+        })
+
+        function createCwlEvents(id: string, count: number): FilteredLogEvents {
+            let events: CloudWatchLogs.FilteredLogEvents = []
+            for (let i = 0; i < count; i++) {
+                events = events.concat({ message: `message-${id}`, logStreamName: `stream-${id}` })
+            }
+            return events
+        }
+
+        async function buildCwlAction(isPage1Empty: boolean): Promise<CloudWatchLogsAction> {
+            return async function (
+                logGroupInfo: CloudWatchLogsGroupInfo,
+                parameters: CloudWatchLogsParameters,
+                nextToken?: CloudWatchLogs.NextToken
+            ) {
+                return getSimulatedCwlResponse(nextToken, isPage1Empty)
+            }
+        }
+
+        /**
+         * This function returns a simulated cloud watch logs reponse for a given token.
+         * @param token The page token
+         * @param isPage1Empty A flag to indicate if Page 1 should have data/isn't the tail.
+         * @returns
+         */
+        function getSimulatedCwlResponse(
+            token: CloudWatchLogs.NextToken | undefined,
+            isPage1Empty: boolean
+        ): CloudWatchLogsResponse {
+            switch (token) {
+                case pageToken1:
+                    if (isPage1Empty) {
+                        return { events: [], nextForwardToken: undefined, nextBackwardToken: pageToken0 }
+                    }
+                    return {
+                        events: createCwlEvents('P1', 1),
+                        nextForwardToken: pageToken2,
+                        nextBackwardToken: pageToken0,
+                    }
+                case pageToken2:
+                    return { events: [], nextForwardToken: undefined, nextBackwardToken: pageToken1 }
+                default: // pageToken0
+                    return {
+                        events: createCwlEvents('P0', 1),
+                        nextForwardToken: pageToken1,
+                        nextBackwardToken: undefined,
+                    }
+            }
+        }
+
+        it('can retrieve new logs when a page is updated', async () => {
+            // Swap to use other paginate testing function
+            const page1IsEmpty = true
+            paginatedData.retrieveLogsFunction = await buildCwlAction(page1IsEmpty)
+
+            // -- Make first call.
+            await registry.updateLog(paginatedUri)
+            const firstUpdatedData = registry.getLogData(paginatedUri)
+            assert(firstUpdatedData)
+
+            const firstActual = {
+                events: firstUpdatedData.data,
+                nextForwardToken: firstUpdatedData.next?.token,
+                nextBackwardToken: firstUpdatedData.previous?.token,
+            }
+            const firstExpected = getSimulatedCwlResponse(pageToken0, page1IsEmpty)
+            assert.deepStrictEqual(firstActual, firstExpected)
```

Comment:
Seems like a pattern that could be de-duped. Maybe merged with `testUpdateLog` somehow?